File: leetcode/Tree/104_Maximum Depth of Binary Tree/ans.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    # @param {TreeNode} root
    # @return {integer}
    def maxDepth(self, root):
    	if root == None:
            return 0
        return max(self.maxDepth(root.left) + 1, self.maxDepth(root.right) + 1);

File: leetcode/Tree/104_Maximum Depth of Binary Tree/ans.c
/**
 * @Time Complexity: O(n)
 */
#include <stdio.h>

/**
 * Definition for singly-linked list.
 */
struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
};

/* Solution */
int maxDepth(struct TreeNode* root) {
	if (NULL == root)
		return 0;
	int ld = maxDepth(root->left) + 1;
	int rd = maxDepth(root->right) + 1;
	return ld > rd ? ld : rd;
}

/* Test Case */
int main() {
}
